cmd/provision/main.go

```diff
@@ -98,74 +113,124 @@ func main() {
 	logger.Error(fmt.Sprintf("Provision service terminated: %s", err))
 }
 
-func startHTTPServer(svc provision.Service, cfg config, logger logger.Logger, errs chan error) {
-	p := fmt.Sprintf(":%s", cfg.httpPort)
-	if cfg.serverCert != "" || cfg.serverKey != "" {
+func startHTTPServer(svc provision.Service, cfg provision.Config, logger logger.Logger, errs chan error) {
+	p := fmt.Sprintf(":%s", cfg.Server.HTTPPort)
+	if cfg.Server.ServerCert != "" || cfg.Server.ServerKey != "" {
 		logger.Info(fmt.Sprintf("Provision service started using https on port %s with cert %s key %s",
-			cfg.httpPort, cfg.serverCert, cfg.serverKey))
-		errs <- http.ListenAndServeTLS(p, cfg.serverCert, cfg.serverKey, api.MakeHandler(svc))
+			cfg.Server.HTTPPort, cfg.Server.ServerCert, cfg.Server.ServerKey))
+		errs <- http.ListenAndServeTLS(p, cfg.Server.ServerCert, cfg.Server.ServerKey, api.MakeHandler(svc))
 		return
 	}
-	logger.Info(fmt.Sprintf("Provision service started using http on port %s", cfg.httpPort))
+	logger.Info(fmt.Sprintf("Provision service started using http on port %s", cfg.Server.HTTPPort))
 	errs <- http.ListenAndServe(p, api.MakeHandler(svc))
 }
 
-func loadConfig() config {
-	tls, err := strconv.ParseBool(mainflux.Env(envClientTLS, defClientTLS))
+func loadConfigFromFile(file string, logger logger.Logger) (provision.Config, error) {
+	_, err := os.Stat(file)
+	if os.IsNotExist(err) {
+		logger.Warn(fmt.Sprintf("Failed to load from file: %v", err))
+		return provision.Config{}, errMissingConfigFile
+	}
+	c, err := provision.Read(file)
 	if err != nil {
-		log.Fatalf(fmt.Sprintf("Cannot determine TLS variable %s", err.Error()))
+		logger.Warn(fmt.Sprintf("Failed to load from file: %v", err))
+		return provision.Config{}, err
+	}
+	return c, nil
+}
+
+func loadConfig() (provision.Config, error) {
```

Comment:
This is not consistent with other services. `loadConfig` is supposed to load environment variables or use default values, nothing else.  Please, split it in multiple functions and don't use the logger.

Reply:
@manuio can you clarify this remark please